nucypher/network/server.py

```diff
@@ -157,18 +154,27 @@ def threshold_decrypt():
             f"Threshold decryption request for ritual ID #{decryption_request.ritual_id}"
         )
 
-        # Deserialize and instantiate ConditionLingo from the request data
-        conditions_data = str(decryption_request.conditions)  # nucypher_core.Conditions -> str
-        lingo = ConditionLingo.from_list(json.loads(conditions_data))  # str -> list -> ConditionLingo
-
         # requester-supplied condition eval context
         context = None
         if decryption_request.context:
-            context = json.loads(str(decryption_request.context)) or dict()  # nucypher_core.Context -> str -> dict
+            context = (
+                json.loads(str(decryption_request.context)) or dict()
+            )  # nucypher_core.Context -> str -> dict
+
+        # obtain condition from request
+        condition_lingo = json.loads(
+            str(decryption_request.conditions)
+        )  # nucypher_core.Conditions -> str -> Lingo
+        if not condition_lingo:
+            # TODO is this needed - this should never happen for CBD - defeats the purpose
```

Comment:
Few thoughts:
- Nothing is stopping the client from breaking the API contract
- Regardless, this is fixable to an extent by `nucypher-ts` (and other clients) by adding input validation
- We can only validate that after decrypting the inputs, so for me it makes sense to check this here. On the other hand, if `condition_lingo` is "falsy" (missing, invalid, and so on) then perhaps we could put this into a catch-all clause in `evaluate_condition_lingo`

Reply:
Yep. The only reason I didn't move this block of code into `evaluate_condition_lingo()` is because we also use that function for PRE-TDec. Once we drop support for pre-tdec then this code can go into `evaluate_condition_lingo()`.